### PR TITLE
Update SymbolKey to use the new CreateAnonymousType api.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide   note that this branches off of https://github.com/dotnet/roslyn/pull/14298
When that goes in, it will make this PR a lot easier to understand.

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-ide This is now much easier to review.  Please take a look when you get a chance.  Thanks!

**Comment by CyrusNajmabadi** [comment]
test windows_debug_unit32_prtest  please

File: src/Workspaces/Core/Portable/SymbolKey/SymbolKey.AnonymousTypeSymbolKey.cs [hunk 1]
```diff
@@ -15,19 +15,25 @@ public static void Create(INamedTypeSymbol symbol, SymbolKeyWriter visitor)
...
+                visitor.WriteLocationArray(propertyLocations);
             }
 
             public static int GetHashCode(GetHashCodeReader reader)
             {
                 // The hash of the underlying type is good enough, we don't need to include names.
                 var symbolKeyHashCode = reader.ReadSymbolKeyArrayHashCode();
-                var elementNames = reader.ReadStringArray();
+                var propertyNames = reader.ReadStringArray();
+                var propertyIsReadOnly = reader.ReadBooleanArray();
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/SymbolKey/SymbolKey.AnonymousTypeSymbolKey.cs]: Note: all the GetHashCode functionality was removed in: https://github.com/dotnet/roslyn/pull/14330

However, to answer your question:

SymbolKeys are just strings.  Therefore, if we're computing anything about them, we need to stream over all the data they have.  If we don't read the data, then the stream-pointer will be in teh wrong place, and that will mean we can't properly read the rest of the data we need.

**Comment by CyrusNajmabadi** [comment]
Preemptively merging in.  Let me know if there are any issues you want me to take care of.

**Comment by jmarolf** [comment]
👍

**Pull request merged**